File: run_migration.py
#!/usr/bin/env python3
"""
Run Supabase migration using Management API
"""
import requests
import json
import sys

# Supabase credentials
SUPABASE_URL = "https://togpvwfxmydgitkwqdgd.supabase.co"
import os
SERVICE_ROLE_KEY = os.getenv('SUPABASE_SERVICE_ROLE_KEY')

if not SERVICE_ROLE_KEY:
    print("❌ Error: SUPABASE_SERVICE_ROLE_KEY environment variable is required")
    print("   Set it in your .env file or export it before running this script")
    exit(1)
PROJECT_REF = "togpvwfxmydgitkwqdgd"

# Read SQL file
try:
    with open('supabase/migrations/002_create_user_profile_trigger.sql', 'r') as f:
        sql_content = f.read()
except FileNotFoundError:
    print("Error: Migration file not found")
    sys.exit(1)

# Use Supabase Management API
# Note: This requires the Management API access token, not the service role key
# For now, we'll use the Supabase Dashboard approach

print("=" * 60)
print("SUPABASE MIGRATION SQL")
print("=" * 60)
print("\nSince Supabase REST API doesn't support direct SQL execution,")
print("please copy and paste this SQL into your Supabase Dashboard:")
print("\n1. Go to: https://supabase.com/dashboard/project/togpvwfxmydgitkwqdgd/sql/new")
print("2. Paste the SQL below")
print("3. Click 'Run'")
print("\n" + "=" * 60)
print(sql_content)
print("=" * 60)

# Alternative: Try using Supabase CLI if linked
print("\n\nAlternatively, if you have Supabase CLI linked to your project:")
print("  supabase db push")

File: run_migration.js
#!/usr/bin/env node
/**
 * Run Supabase migration using service role key
 * This script executes the SQL migration directly
 */

const fs = require('fs');
const https = require('https');

const SUPABASE_URL = process.env.SUPABASE_URL || 'https://togpvwfxmydgitkwqdgd.supabase.co';
// Note: This script doesn't actually use the service role key
// It just displays SQL to run manually in Supabase Dashboard

// Read SQL file
const sqlContent = fs.readFileSync('supabase/migrations/002_create_user_profile_trigger.sql', 'utf8');

// Supabase doesn't have a direct SQL execution endpoint via REST API
// We need to use the Management API or run it manually
console.log('='.repeat(60));
console.log('SUPABASE MIGRATION SQL');
console.log('='.repeat(60));
console.log('\n⚠️  Supabase REST API does not support direct SQL execution.');
console.log('\n📋 Please run this SQL manually in Supabase Dashboard:');
console.log('\n1. Go to: https://supabase.com/dashboard/project/togpvwfxmydgitkwqdgd/sql/new');
console.log('2. Copy and paste the SQL below');
console.log('3. Click "Run" button');
console.log('\n' + '='.repeat(60));
console.log(sqlContent);
console.log('='.repeat(60));

// Alternative: Try using Supabase CLI
console.log('\n\n💡 Alternative: Use Supabase CLI');
console.log('   1. Install Supabase CLI: npm install -g supabase');
console.log('   2. Link project: supabase link --project-ref togpvwfxmydgitkwqdgd');
console.log('   3. Run migration: supabase db push');
console.log('\n   Or use psql with connection string from Supabase Dashboard');

